Formula/mupdf.rb

```diff
@@ -21,23 +22,55 @@ class Mupdf < Formula
   end
 
   depends_on "pkg-config" => :build
-  depends_on "freeglut"
-  depends_on "mesa"
+  depends_on "freetype"
+  depends_on "gumbo-parser"
+  depends_on "harfbuzz"
+  depends_on "jbig2dec"
+  depends_on "jpeg-turbo"
+  depends_on "mujs"
+  depends_on "openjpeg"
+  depends_on "openssl@1.1"
+
+  uses_from_macos "zlib"
+
+  on_linux do
+    depends_on "freeglut"
+    depends_on "libx11"
+    depends_on "libxext"
+    depends_on "mesa"
+  end
 
   conflicts_with "mupdf-tools",
     because: "mupdf and mupdf-tools install the same binaries"
 
   def install
-    glut_cflags = `pkg-config --cflags glut gl`.chomp
-    glut_libs = `pkg-config --libs glut gl`.chomp
-    system "make", "install",
-           "build=release",
-           "shared=yes",
-           "verbose=yes",
-           "CC=#{ENV.cc}",
-           "SYS_GLUT_CFLAGS=#{glut_cflags}",
-           "SYS_GLUT_LIBS=#{glut_libs}",
-           "prefix=#{prefix}"
+    # Remove bundled libraries excluding `extract` and "strongly preferred" `lcms2mt` (lcms2 fork)
+    keep = %w[extract lcms2]
+    (buildpath/"thirdparty").each_child { |path| path.rmtree if keep.exclude? path.basename.to_s }
+
+    args = %W[
+      build=release
+      shared=yes
+      verbose=yes
+      prefix=#{prefix}
+      CC=#{ENV.cc}
+      USE_SYSTEM_LIBS=yes
+      USE_SYSTEM_MUJS=yes
+    ]
+    # Build only runs pkg-config for libcrypto on macOS, so help find other libs
+    if OS.mac?
+      [
+        ["FREETYPE", "freetype2"],
+        ["GUMBO", "gumbo"],
+        ["HARFBUZZ", "harfbuzz"],
+        ["LIBJPEG", "libjpeg"],
+        ["OPENJPEG", "libopenjp2"],
+      ].each do |argname, libname|
```

Comment:
Seems much more natural as a hash.
```suggestion
      {
        FREETYPE: "freetype2",
        GUMBO:    "gumbo",
        HARFBUZZ: "harfbuzz",
        LIBJPEG:  "libjpeg",
        OPENJPEG: "libopenjp2",
      }.each do |argname, libname|
```